examples/assets-and-custom-transfer-logic/assets/poi-approval.teal

```diff
@@ -0,0 +1,308 @@
+int axfer
+==
+&&
+gtxn 2 TypeEnum
+int pay
+==
+&&
+// Verify that this is index the stateful contract call
+txn GroupIndex
+int 0
+==
+&&
+bz failed
+
+// check no rekeying etc
+gtxn 0 RekeyTo
+global ZeroAddress
+==
+gtxn 1 RekeyTo
+global ZeroAddress
+==
+&&
+gtxn 2 RekeyTo
+global ZeroAddress
+==
+&&
+gtxn 0 CloseRemainderTo
+global ZeroAddress
+==
+&&
+gtxn 1 CloseRemainderTo
+global ZeroAddress
+==
+&&
+gtxn 2 CloseRemainderTo
+global ZeroAddress
+==
+&&
+gtxn 0 AssetCloseTo
+global ZeroAddress
+==
+&&
+gtxn 1 AssetCloseTo
+global ZeroAddress
+==
+&&
+gtxn 2 AssetCloseTo
+global ZeroAddress
+==
+&&
+bz failed
+
+// tx 0 - check level smart contract call - signed by sender of asset
+// tx 1 - clawback transactions that moves the frozen asset from sender to receiver - signed by clawback-escrow
+// tx 2 - payment transaction from sender to clawback-escrow to pay for the fee of the clawback
+
+// get asset id from global
+int 0 //current app
+byte "AssetID"
+app_global_get_ex
+bz failed
+// store assetid
+store 11
+
+// verify first transaction
+// check level smart contract call - signed by asset sender
+gtxn 0 ApplicationID
+global CurrentApplicationID
+==
+gtxn 0 Sender
+gtxn 2 Sender
+==
+&&
+gtxn 0 Sender
+gtxn 1 AssetSender
+==
+&&
+bz failed
+
+// verify second transaction
+// tx 1 - clawback transactions that moves the frozen asset from sender to receiver - signed by clawback-escrow
+// verify the account sent in the accounts array is
+// Note: the receiver of the asset is asset xfer
+gtxn 0 Accounts 1
+gtxn 1 AssetReceiver
+==
+gtxn 1 XferAsset
+load 11
+==
+&&
+bz failed
+
+// verify third transaction
+// tx 2 - payment transaction from sender to clawback-escrow to pay for the fee of the clawback
+
+// verify the fee amount is good
+gtxn 2 Amount
+gtxn 1 Fee
+>=
+
+bz failed
+
+
+// now handle the level check
+// get asset sender stored accreditation level
+int 0 //sender
+gtxn 0 ApplicationID
+byte "Accred-Level"
+app_local_get_ex
+bz failed // must have an accred level set
+
+// top of the stack has the current user level
+// load asset required level
+byte "AssetLevel"
+app_global_get
+>=
+bz failed // users level not high enough
+
+// get asset receiver stored accreditation level
+int 1 //asset receiver account
+txn ApplicationID
+byte "Accred-Level"
+app_local_get_ex
+bz failed // must have an accred level set
+// top of the stack has the current user level
+// load asset required level
+byte "AssetLevel"
+app_global_get
+>=
+bz failed // users level not high enough
+
+int 1
+return
+
+handle_optin:
+txn NumAppArgs
+int 0
+==
+txn GroupIndex
+int 0
+==
+&&
+bz failed
+int 1
+return
+handle_closeout:
+txn GroupIndex
+int 0
+==
+bz failed
+// we should not have to clear the
+// accred level
+int 1
+return
+
+// By default, disallow updating or deleting the app. Add custom authorization
+// logic below to allow updating or deletion in certain circumstances.
+handle_updateapp:
+handle_deleteapp:
+err
+failed:
+int 0
+return
```

Comment:
I converted the other to PyTEAL because in that stateless smart contract, we required `assetId`, `appId` in the contract itself. So instead of manually adding values each time there, it was better to use PyTEAL and pass those via `Params`. eg.
```
  const escrowParams = {
    ASSET_ID: assetInfo.assetIndex,
    APP_ID: appInfo.appID
  };
  const escrowLsig = await deployer.loadLogic('clawback-escrow.py', [], escrowParams);
  ```
  
  These scripts were taken from here: [https://github.com/algorand/smart-contracts/blob/master/devrel/poi/poi.teal](https://github.com/algorand/smart-contracts/blob/master/devrel/poi/poi.teal). 
  Please let me know you think this should be converted to PyTEAL as well. Will do it.

Reply:
It's fine. Please add a comment in the TEAL code with a link to the original source. In the README we can also add a link to https://github.com/algorand/smart-contracts/blob/master/devrel

Reply:
Adding comment in TEAL code. In README the link is already added in `## More Information` section